firewood/src/proof.rs

```diff
@@ -209,7 +209,7 @@ impl<T: Hashable> Proof<T> {
             return Err(ProofError::Empty);
         };
 
-        #[allow(clippy::useless_conversion)]
+        #[cfg_attr(not(feature = "ethhash"), expect(clippy::useless_conversion))]
```

Comment:
I better way to fix this is to conditionally compile this let statement to not even do the into() when ethhash is enabled.

Reply:
Yeah, I did that but it was rather ugly. I think the new approach is a good middle ground.